### ISSUE TITLE
launch chrome attach to wrong tab

### ISSUE THREAD

**Issue by gertcuykens** [Microsoft/vscode-chrome-debug#79]
When you launch chrome in debug mode, it opens a new tab but vcode attaches to soon to the already opened tab that can be any website. It should attach to the tab it created instead. Or look for the url in current tabs before opening a new one.

```json
{
	"version": "0.10.6",
	"configurations": [
        {
            "name": "Launch chrome",
            "type": "chrome",
            "request": "launch",
            "url": "http://localhost:8080/",
            "webRoot": "default",
            "sourceMaps": true
        }
	]
}
```

**Comment by roblourens** [comment]
It does look for the url in current tabs, with a timeout, but the timeout may be too short here. You can work around it by setting the `userDataDir` property to a temp directory, so Chrome starts in a new instance instead of a new tab.

**Comment by gertcuykens** [comment]
No the last thing I want is a new window every time I restart my debug :) It should only open anything when it doesn't find the url. Can you please keep it open until timeout is fixed in master?

**Comment by roblourens** [comment]
Interesting, I'll take a closer look

**Comment by auchenberg** [comment]
Does this happen because you have multiple tabs open with `http://localhost:8080/`? @gertcuykens

**Comment by gertcuykens** [comment]
Happens with one localhost tab or multiple localhost tabs, PS doesn't it happen on your system? Could be mac os specific then or chrome version specific

**Comment by mykohsu** [comment]
I have the same issue.

For example, tab #1 is open with the launch URL.
Launching debug opens tab #2, but debugger attaches to tab #1.

Or, tab #1 is open with non-matching URL (such as this github issue page).
Launching debug opens tab #2, but debugger then attaches to tab #1???

**Comment by vkniazeu** [comment]
I'm experiencing the same issue on Windows 10.
It feels like the "launch" configuration tries to attach the debugger before the new needed tab is opened.
I'm getting warnings about the debugger trying to attach to other open tabs.
If there's already a tab with this URL it will find it, but you will need to refresh the page.
Even this works sporadically.
What seems to work is this very redundant flow:
- run the "launch" configuration
- let it open the relevant tab than debugger needs it
-  run the "attach" configuration
- reload the tab manually
This is the only flow I have found to work so far.
Thanks!

**Comment by roblourens** [comment]
Fixed in 0.4.8! Give it a shot and let me know if any issues.
